Common/RoguelikeChange/RoguelikeItemOverhaulData.cs

```diff
@@ -0,0 +1,207 @@
+		ArmorSetBonusToolTip = "Increase defense by 7" +
+						"\nYour attack can inflict irradiation poison" +
+						"\nLead irradiation increase enemy defense by 20 but deal 50 DoT";
+	}
+}
+public class CopperArmorSet : ArmorSet {
+	public CopperArmorSet() : base(ItemID.CopperHelmet, ItemID.CopperChainmail, ItemID.CopperGreaves) {
+		ArmorSetBonusToolTip = "Increase movement speed by 15%" +
+					   "\nEvery 50 hit to enemy grants you the over charged for 5s" +
+					   "\nDuring the rain, your hit requirement reduce by half" +
+					   "\nOver charged: Increases movement speed, weapon speed, damage by 10% and create a aura that damage enemy";
+	}
+}
+public class PearlwoodSet : ArmorSet {
+	public PearlwoodSet() : base(ItemID.PearlwoodHelmet, ItemID.PearlwoodBreastplate, ItemID.PearlwoodGreaves) {
+		ArmorSetBonusToolTip = "Increase movement speed by 35%" +
+						"\nAttacking an enemy summons 6 hallow Swords that deals 5 damage with 4 seconds cooldown" +
+						"\nIncrease damage by 15% during day" +
+						"\nIncrease defense By 12" +
+						"\nWhen in Hallow biome:" +
+						"\n Hallow Swords deal 15 damage";
+	}
+}
+public class IronArmorSet : ArmorSet {
+	public IronArmorSet() : base(ItemID.IronHelmet, ItemID.IronChainmail, ItemID.IronGreaves) {
+		ArmorSetBonusToolTip = "Increase damage reduction 2.5%" +
+					   "\nIncrease defense effectivness by 10%" +
+					   "\nIncrease damage by 5%" +
+					   "\nDecrease movement speed 5%" +
+					   "\nWhile under 50% HP, gain 15 bonus defense, but -10% less attack speed";
+	}
+}
+public class SilverArmorSet : ArmorSet {
+	public SilverArmorSet() : base(ItemID.SilverHelmet, ItemID.SilverChainmail, ItemID.SilverGreaves) {
+		ArmorSetBonusToolTip = "During the day :" +
+					   "\nGain 10 defense" +
+					   "\nDuring the night :" +
+					   "\nIncrease damage by 10%" +
+					   "\nAt full life, these effects are multiply by 2";
+	}
+}
+public class TungstenArmorSet : ArmorSet {
+	public TungstenArmorSet() : base(ItemID.TungstenHelmet, ItemID.TungstenChainmail, ItemID.TungstenGreaves) {
+		ArmorSetBonusToolTip = "Increase defense by 15" +
+					   "\nWhen at full hp :" +
+					   "\nReduce your defense down to 0" +
+					   "\nIncrease speed by 30%" +
+					   "\nThe closer your enemy is, the more damage increases";
+	}
+}
+public class GoldArmorSet : ArmorSet {
+	public GoldArmorSet() : base(ItemID.GoldHelmet, ItemID.GoldChainmail, ItemID.GoldGreaves) {
+		ArmorSetBonusToolTip = "Your attack have 15% to inflict Midas for 10 seconds" +
+					   "\nAttacking enemies with midas debuff will : " +
+					   "\nDeal additional damage based on their defense";
+	}
+}
+public class PlatinumArmorSet : ArmorSet {
+	public PlatinumArmorSet() : base(ItemID.PlatinumHelmet, ItemID.PlatinumChainmail, ItemID.PlatinumGreaves) {
+		ArmorSetBonusToolTip = "Increase weapon uses speed by 35%" +
+					   "\nAttacking too much will lit on fire";
+	}
+}
+public class ArmorSet {
```

Comment:
Make this class implement `ILoadable` or inherit `ModType`, add `abstract` modifier to class so it doesn't attempts to load itself, override `Register` method and seal it then, and call `ModTypeLookup<ArmorSet>.Register(this)` in there

Reply:
It is suppose to only hold data and process it, but I think I will try and make a localize file instead